### PR TITLE
Fixed release notes bug

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/19904

## Screenshots
Before the fix: 
image

After the fix:
image

### CODE REVIEW

**Comment by Shellyber** [comment]
Doing some more tests.

File: demisto_sdk/commands/validate/file_validator.py [hunk 1]
```diff
@@ -392,6 +374,7 @@ def validate_modified_files(self, modified_files):  # noqa: C901
 
             elif 'CHANGELOG' in file_path:
                 self.is_valid_release_notes(file_path)
+                continue
```
Comment (Itay4) [inline demisto_sdk/commands/validate/file_validator.py:377]: why is that `continue` needed? 
if im not missing something, it would continue to the next iteration anyway

File: demisto_sdk/commands/validate/file_validator.py [hunk 2]
```diff
@@ -503,11 +486,12 @@ def validate_added_files(self, added_files, file_type: str = None):  # noqa: C90
 
             elif checked_type(file_path, JSON_ALL_INCIDENT_TYPES_REGEXES):
                 incident_type_validator = IncidentTypeValidator(structure_validator)
-                if not incident_type_validator.is_valid_incident_type(validate_rn=not file_type):
+                if not incident_type_validator.is_valid_incident_type(validate_rn=False):
                     self._is_valid = False
 
             elif 'CHANGELOG' in file_path:
                 self.is_valid_release_notes(file_path)
+                continue
```
Comment (Itay4) [inline demisto_sdk/commands/validate/file_validator.py:494]: why is that `continue` needed? 
if im not missing something, it would continue to the next iteration anyway

**Comment by Shellyber** [comment]
Build that failed before the fix:
https://circleci.com/gh/demisto/content/46861
Build that passed after the fix:
https://circleci.com/gh/demisto/content/46865?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: demisto_sdk/commands/validate/file_validator.py (hunk 1 above)
Comment (Shellyber) [reply demisto_sdk/commands/validate/file_validator.py:377]: Forgot to remove. will do

**Comment by Shellyber** [comment]
For the reputation, layout and dashboard according to @yaakovi it's mandatory. must pass even empty release notes

**Comment by Itay4** [comment]
> For the reputation, layout and dashboard according to @yaakovi it's mandatory. must pass even empty release notes

@yaakovi why?

**Comment by yaakovi** [comment]
> > For the reputation, layout and dashboard according to @yaakovi it's mandatory. must pass even empty release notes
> 
> @yaakovi why?

all content entities have a description/comment that specifies their purpose.
checked now: actually, only layout & classification must have changelogs when added.

**Comment by Shellyber** [comment]
@Itay4 @yaakovi Done

**Comment by Itay4** [comment]
@Shellyber lets do the following: 
1. open issue for `release_notes.py` script in the content repo to not expect rn for layout & classification
2. add TODO with reference to the issue in the code here for layout & classification to set `validate_rn` to `False` after issue is fixed

also, need to add tests here

**Comment by Shellyber** [comment]
@Itay4 @yaakovi Please review again. 
All done

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 3]
```diff
@@ -80,13 +80,14 @@ def is_valid_file(self, validate_rn: bool = True) -> bool:
         ]
         return all(answers)
 
-    def is_valid_beta_integration(self) -> bool:
+    def is_valid_beta_integration(self, validate_rn=True):
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:83]:
```suggestion
    def is_valid_beta_integration(self, validate_rn: bool = True) -> bool:
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 4]
```diff
@@ -80,13 +80,14 @@ def is_valid_file(self, validate_rn: bool = True) -> bool:
...
         """Check whether the beta Integration is valid or not, update the _is_valid field to determine that"""
         answers = [
+            super().is_valid_file(validate_rn),
             self.is_valid_default_arguments(),
             self.is_valid_beta(),
             self.is_valid_image(),
-            self.is_valid_description(beta_integration=True),
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:89]: revert this change

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 5]
```diff
@@ -80,13 +80,14 @@ def is_valid_file(self, validate_rn: bool = True) -> bool:
         ]
         return all(answers)
 
-    def is_valid_beta_integration(self) -> bool:
+    def is_valid_beta_integration(self, validate_rn=True):
         """Check whether the beta Integration is valid or not, update the _is_valid field to determine that"""
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:84]: add argument and return type to docstring

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 6]
```diff
@@ -8,18 +8,20 @@
 class PlaybookValidator(BaseValidator):
     """PlaybookValidator is designed to validate the correctness of the file structure we enter to content repo."""
 
-    def is_valid_playbook(self, is_new_playbook=True):  # type: (bool) -> bool
+    def is_valid_playbook(self, is_new_playbook=True, validate_rn=True):  # type: (bool) -> bool
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/playbook.py:11]:
```suggestion
    def is_valid_playbook(self, is_new_playbook: bool = True, validate_rn: bool = True) -> bool: 
```

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 7]
```diff
@@ -1,8 +1,10 @@
+# isort:skip_file
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:1]: why do you skip isort?

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 8]
```diff
@@ -257,3 +260,20 @@ def test_is_all_params_not_hidden(self, source, answer):
...
+        (VALID_INCIDENT_FIELD_PATH, INCIDENT_FIELD_TARGET),
+        (VALID_TEST_PLAYBOOK_PATH, PLAYBOOK_TARGET)
+    ]
+
+    @pytest.mark.parametrize('source, target', INPUTS_RELEASE_NOTES_VALIDATION)
+    def test_is_file_valid_rn(self, source, target):
+        # type: (str, str) -> None
+        try:
+            copyfile(source, target)
+            assert FilesValidator(validate_conf_json=False).is_valid_structure()
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:277]: this just validates schema, and doesnt actually get to the release notes validator 
what we need to test here is `validate_added_files`

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 9]
```diff
@@ -257,3 +260,20 @@ def test_is_all_params_not_hidden(self, source, answer):
         structure = StructureValidator(source)
         validator = IntegrationValidator(structure)
         assert validator.is_all_params_not_hidden() is answer
+
+    INPUTS_RELEASE_NOTES_VALIDATION = [
+        (VALID_INTEGRATION_TEST_PATH, INTEGRATION_TARGET),
+        (VALID_SCRIPT_PATH, SCRIPT_TARGET),
+        (VALID_DASHBOARD_PATH, DASHBOARD_TARGET),
+        (VALID_INCIDENT_FIELD_PATH, INCIDENT_FIELD_TARGET),
+        (VALID_TEST_PLAYBOOK_PATH, PLAYBOOK_TARGET)
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:269]: need also to check the following: incident type, reputation, beta integration

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 8 above)
Comment (Shellyber) [reply demisto_sdk/commands/validate/tests/validators_test.py:277]: it does. it tests the structure and then goes to valid file and tests everything. If there is a valid new file and it has release notes it will fail

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 7 above)
Comment (Shellyber) [reply demisto_sdk/commands/validate/tests/validators_test.py:1]: Cause, I fix what it wanted and still failed on the same file

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 8 above)
Comment (Itay4) [reply demisto_sdk/commands/validate/tests/validators_test.py:277]: unless im missing something, for every test case it runs `validate_all_files`, which does not validate release notes, but only scheme

Comment (Shellyber) [reply demisto_sdk/commands/validate/tests/validators_test.py:282]: We don't need the releas notes validator cause I'm not interested if the release notes file is valid (which is not cause it doesn't exists).
The `is_valid_structure` initiates the `self.validate_added_files({self.file_path}, file_type=find_type(self.file_path))` that check all the added files including the rn validation

Comment (Itay4) [reply demisto_sdk/commands/validate/tests/validators_test.py:282]: `is_valid_structure` calls `validate_added_files` only if `file_path` is provided, but it is not in your case, so it will always continue to the `else` clause and run `validate_all_files`

**Comment by Shellyber** [comment]
@Itay4 check please

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 10]
```diff
@@ -274,11 +277,62 @@ def test_is_all_params_not_hidden(self, source, answer):
...
             assert FilesValidator(validate_conf_json=False).is_valid_structure()
         finally:
             os.remove(target)
+
+    @staticmethod
+    def mock_get_diff():
+        return None
+
+    @staticmethod
+    def mock_structure_validator():
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:293]: remove this func

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 11]
```diff
@@ -274,11 +277,62 @@ def test_is_all_params_not_hidden(self, source, answer):
...
+    def test_is_file_structure(self, source, target):
         # type: (str, str) -> None
         try:
             copyfile(source, target)
             assert FilesValidator(validate_conf_json=False).is_valid_structure()
         finally:
             os.remove(target)
+
+    @staticmethod
+    def mock_get_diff():
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:289]: remove this func

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 12]
```diff
@@ -274,11 +277,62 @@ def test_is_all_params_not_hidden(self, source, answer):
...
+        ([VALID_INTEGRATION_TEST_PATH], 'integration'),
+        ([VALID_TEST_PLAYBOOK_PATH], 'playbook'),
+        ([VALID_DASHBOARD_PATH], 'dashboard'),
+        ([VALID_INCIDENT_FIELD_PATH], 'incidentfield'),
+        ([VALID_REPUTATION_PATH], 'reputation')
+    ]
+
+    @pytest.mark.parametrize('file_path, file_type', FILE_PATHS)
+    def test_is_valid_rn(self, mocker, file_path, file_type):
+        mocker.patch.object(ReleaseNotesValidator, 'get_master_diff', side_effect=self.mock_get_diff)
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:306]:
```suggestion
        mocker.patch.object(ReleaseNotesValidator, 'get_master_diff', side_effect=None)
```

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 13]
```diff
@@ -257,3 +264,75 @@ def test_is_all_params_not_hidden(self, source, answer):
...
+    @staticmethod
+    def mock_structure_validator():
+        return True
+
+    FILE_PATHS = [
+        ([VALID_INTEGRATION_TEST_PATH], 'integration'),
+        ([VALID_TEST_PLAYBOOK_PATH], 'playbook'),
+        ([VALID_DASHBOARD_PATH], 'dashboard'),
+        ([VALID_INCIDENT_FIELD_PATH], 'incidentfield'),
+        ([VALID_REPUTATION_PATH], 'reputation')
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:301]: add beta integration and incident type to test

File: demisto_sdk/commands/common/hook_validations/playbook.py [hunk 14]
```diff
@@ -8,18 +8,20 @@
 class PlaybookValidator(BaseValidator):
     """PlaybookValidator is designed to validate the correctness of the file structure we enter to content repo."""
 
-    def is_valid_playbook(self, is_new_playbook=True):  # type: (bool) -> bool
+    def is_valid_playbook(self, is_new_playbook: bool = True, validate_rn: bool = True):  # type: (bool) -> bool
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/playbook.py:11]:
```suggestion
    def is_valid_playbook(self, is_new_playbook: bool = True, validate_rn: bool = True) -> bool:
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 15]
```diff
@@ -82,9 +82,16 @@ def is_valid_file(self, validate_rn: bool = True) -> bool:
         ]
         return all(answers)
 
-    def is_valid_beta_integration(self) -> bool:
-        """Check whether the beta Integration is valid or not, update the _is_valid field to determine that"""
+    def is_valid_beta_integration(self, validate_rn: bool = True):
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:85]:
```suggestion
    def is_valid_beta_integration(self, validate_rn: bool = True) -> bool:
```

**Comment by Shellyber** [comment]
@Itay4 Already triggered build with this SDK branch. See at the beginning:
Build that passed after the fix:
https://circleci.com/gh/demisto/content/46865?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by Shellyber** [comment]
@Itay4 Done

**Comment by Itay4** [comment]
> @Itay4 Already triggered build with this SDK branch. See at the beginning:
> Build that passed after the fix:
> https://circleci.com/gh/demisto/content/46865?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

what i meant is that we need to check that the validation step content build passes as expected with the change here, to make sure we didnt break anything, not to check if what you did works

**Comment by Shellyber** [comment]
@Itay4 https://circleci.com/gh/demisto/content/47337
So far so good
